docs/demography.md

```diff
@@ -461,71 +468,134 @@ demography.debug()
 generation_time = 25
-T_AF = 220e3 / generation_time
-T_B = 140e3 / generation_time
-T_EU_AS = 21.2e3 / generation_time
-# We need to work out the starting population sizes based on the growth
-# rates provided for these two populations
-r_EU = 0.004
-r_AS = 0.0055
-N_EU = N_EU0 / math.exp(-r_EU * T_EU_AS)
-N_AS = N_AS0 / math.exp(-r_AS * T_EU_AS)
-# Migration rates during the various epochs.
-m_AF_B = 25e-5
-m_AF_EU = 3e-5
-m_AF_AS = 1.9e-5
-m_EU_AS = 9.6e-5
-# Population IDs correspond to their indexes in the popupulation
-# configuration array. Therefore, we have 0=YRI, 1=CEU and 2=CHB
-# initially.
+T_OOA = 21.2e3 / generation_time
+T_AMH = 140e3 / generation_time
+T_ANC = 220e3 / generation_time
+# We need to work out the starting (diploid) population sizes based on
+# the growth rates provided for these two populations
+r_CEU = 0.004
+r_CHB = 0.0055
+N_CEU = 1000 / math.exp(-r_CEU * T_OOA)
+N_CHB = 510 / math.exp(-r_CHB * T_OOA)
 
 populations = [
-    msprime.Population(name="YRI", initial_size=N_AF),
-    msprime.Population(name="CEU", initial_size=N_EU, growth_rate=r_EU),
-    msprime.Population(name="CHB", initial_size=N_AS, growth_rate=r_AS),
-]
-migration_matrix = [
-    [0, m_AF_EU, m_AF_AS],
-    [m_AF_EU, 0, m_EU_AS],
-    [m_AF_AS, m_EU_AS, 0],
-]
-events = [
-    # CEU and CHB merge into B with rate changes at T_EU_AS
-    msprime.MassMigration(time=T_EU_AS, source=2, destination=1, proportion=1.0),
-    msprime.MigrationRateChange(time=T_EU_AS, rate=0),
-    msprime.MigrationRateChange(time=T_EU_AS, rate=m_AF_B, matrix_index=(0, 1)),
-    msprime.MigrationRateChange(time=T_EU_AS, rate=m_AF_B, matrix_index=(1, 0)),
-    msprime.PopulationParametersChange(
-        time=T_EU_AS, initial_size=N_B, growth_rate=0, population_id=1
+    msprime.Population(
+        name="YRI",
+        description="Yoruba in Ibadan, Nigeria",
+        initial_size=12300,
+    ),
+    msprime.Population(
+        name="CEU",
+        description=(
+            "Utah Residents (CEPH) with Northern and Western European Ancestry"
+        ),
+        initial_size=N_CEU,
+        growth_rate=r_CEU,
+    ),
+    msprime.Population(
+        name="CHB",
+        description="Han Chinese in Beijing, China",
+        initial_size=N_CHB,
+        growth_rate=r_CHB,
+    ),
+    msprime.Population(
+        name="OOA",
+        description="Bottleneck out-of-Africa population",
+        initial_size=2100,
     ),
-    # Population B merges into YRI at T_B
-    msprime.MassMigration(time=T_B, source=1, destination=0, proportion=1.0),
-    msprime.MigrationRateChange(time=T_B, rate=0),
-    # Size changes to N_A at T_AF
-    msprime.PopulationParametersChange(
-        time=T_AF, initial_size=N_A, population_id=0
+    msprime.Population(
+        name="AMH", description="Anatomically modern humans", initial_size=12300
+    ),
+    msprime.Population(
+        name="ancestral",
+        description="Equilibrium/root population",
+        initial_size=7300,
```

Comment:
One thing that bugs me is that this `initial_size` applies to `time=0`, yet the intent is that this population doesn't exist until `time=T_ANC`. And the demography debugger output reinforces this confusion. Furthermore, it's possible to sample from this population at `time=0` too, which is almost certainly not desirable.

Reply:
I agree - I'm thinking of adding ``start_time`` and ``end_time`` to Populations here like a-la Demes. I think I'll merge this much first though, stop things getting out of hand.